app/queues/dns/add-record-flow.server.ts

```diff
@@ -0,0 +1,105 @@
+import { FlowProducer } from 'bullmq';
+import { redis } from '~/lib/redis.server';
+import { buildDomain } from '~/utils';
+import { dnsUpdateQueueName } from './workers/dns-update-worker.server';
+import { checkDnsStatusQueueName } from './workers/poll-dns-status-worker.server';
+import { syncDbStatusQueueName } from './workers/sync-db-status-worker.server';
+import { createUserRecord } from '~/models/record.server';
+
+import type { FlowJob } from 'bullmq';
+import type { Record } from '@prisma/client';
+import type { DnsUpdaterData } from './workers/dns-update-worker.server';
+import type { DbRecordSynchronizerData } from './workers/sync-db-status-worker.server';
+
+export type Name = { subdomain: string } | { fqdn: string };
```

Comment:
`Name` is really generic, can we do better?  `DomainName`, or `DomainNameComponent` or something, which could be a `subdomain` or `fqdn`?  Doesn't have to be that, but `Name` is not going to work.  @dadolhay will have thoughts.

Reply:
It's been one of the most difficult tasks for me to name things related to domain. I would like to suggest `DomainType` as it's either partial domain of full domain.

Reply:
> "There are only two hard things in Computer Science: cache invalidation and naming things." --Phil Karlton (Netscape)

You're a true programmer when you hit the hardest problems in CS. Welcome!

The more you can avoid needing to name things, the fewer names you have to come up with.

I don't think `DomainType` is wrong, but `DomainName` or even `Domain` are better imho, since both a `subdomain` and a `fully qualified domain name` are both `domains`.

Reply:
I'd drobably name this `FqdnOrSubdomain`

Reply:
and yes, separating stuff, race conditions, flaky tests, finding others` [monkey patches](https://en.wikipedia.org/wiki/Monkey_patch) and naming stuff are some of the hardest things in sw development

Reply:
There could be many different opinions, but to me `FqdnOrSubdomain` makes sense a lot. Thanks for the suggestion.